src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -2622,11 +2622,24 @@ protected internal bool At(CSharpKeyword keyword)
                 result.Value == keyword;
         }
 
-        protected static Func<SyntaxToken, bool> IsSpacingToken(bool includeNewLines, bool includeComments)
-        {
-            return token => token.Kind == SyntaxKind.Whitespace ||
-                          (includeNewLines && token.Kind == SyntaxKind.NewLine) ||
-                          (includeComments && token.Kind == SyntaxKind.CSharpComment);
+        protected static bool IsSpacingToken(SyntaxToken token)

```

Comment:
Don't these all result in allocations now? `ReadWhile(IsSpacingTokenIncludingNewLinesAndComments)` should allocate a new `Func<bool, SyntaxToken>` delegate object each time its called.

Reply:
I'm not clear on the question. The old code allocated a new Func on each invocation. The new code defined a new function for each NewLines/Comments combination and called that directly instead of allocating the func and doing the newline/comments checks.

---
In reply to: [423924922](https://github.com/dotnet/aspnetcore-tooling/pull/1874#discussion_r423924922) [](ancestors = 423924922)

Reply:
OK, I guess you made it better but not all the way better 😆 

The pattern of `ReadWhile(IsSpacingToken)` where `IsSpacingToken` is a method will allocate a delegate object every time. Yes. It's dumb. 

You can optimize it out by creating the delegate once and caching it in a field. Or you can turn these into lambdas that initialize a delegate in a field.

https://stackoverflow.com/questions/53208516/delegate-instance-allocation-with-method-group-compared-to

Reply:
Your're right, not what I was expected from using a method group. Hopefully addressed in https://github.com/dotnet/aspnetcore-tooling/pull/1896